framework/OutStreams/PlotInterfaces/PlotInterface.py

```diff
@@ -0,0 +1,113 @@
+# Copyright 2017 Battelle Energy Alliance, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Created on April 1, 2021
+
+@author: talbpaul
+"""
+
+from OutStreams import OutStreamInterface
+
+class PlotInterface(OutStreamInterface):
+  """
+    Archetype for Plot implementations
+  """
+  @classmethod
+  def getInputSpecification(cls):
+    """
+      Method to get a reference to a class that specifies the input data for class "cls".
+      @ In, cls, the class for which we are retrieving the specification
+      @ Out, inputSpecification, InputData.ParameterInput, class to use for specifying the input of cls.
+    """
+    spec = super().getInputSpecification()
+    return spec
+
+  def __init__(self):
+    """
+      Init of Base class
+      @ In, None
+      @ Out, None
+    """
+    super().__init__()
+    self.printTag = 'PlotInterface'
+    self.overwrite = True     # overwrite existing creations? # TODO not in input specs?
+    self.subDirectory = None  # directory to save generated files to # TODO should be in spec
+    self.filename = ''        # target file name? TODO should be in spec
```

Comment:
This variables already defined in OutStreamInterface base class, right?

Reply:
True, removed. Thanks.